app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -562,6 +563,71 @@ def vm_compute_attributes(vm)
       vm_attrs
     end
 
+    def normalize_vm_attrs(vm_attrs)
+      normalized = slice_vm_attributes(vm_attrs, ['cpus', 'firmware', 'guest_id', 'annotation', 'resource_pool_id', 'image_id'])
+
+      normalized['cores_per_socket'] = vm_attrs['corespersocket']
+      normalized['memory'] = vm_attrs['memory_mb'].nil? ? nil : (vm_attrs['memory_mb'].to_i * 1024)
+
+      normalized['folder_path'] = vm_attrs['path']
+      normalized['folder_name'] = self.folders.detect { |f| f.path == normalized['folder_path'] }.try(:name)
+
+      normalized['cluster_id'] = self.available_clusters.detect { |c| c.name == vm_attrs['cluster'] }.try(:id)
+      normalized['cluster_name'] = vm_attrs['cluster']
+      normalized['cluster_name'] = nil if normalized['cluster_name'].empty?
+
+      if normalized['cluster_name']
+        normalized['resource_pool_id'] = self.resource_pools(:cluster_id => normalized['cluster_name']).detect { |p| p.name == vm_attrs['resource_pool'] }.try(:id)
+      end
+      normalized['resource_pool_name'] = vm_attrs['resource_pool']
+      normalized['resource_pool_name'] = nil if normalized['resource_pool_name'].empty?
+
+      normalized['guest_name'] = self.guest_types[vm_attrs['guest_id']]
```

Comment:
I think the guest_name parameter should be the real name (or label) of the guest_id. And this parameter should be named `guest_id`.

Reply:
Are you sure with `guest_id`? I'm not sure I understand your comment, but `guest_id` gets normally copied over and `guest_name` contains description from the `guest_types()` mapping.

Reply:
I'm not sure, anymore. I guess I must have misread the code. Carry on! :-)